warehouse/query-core/src/main/java/datawave/query/data/parsers/FieldIndexKey.java

```diff
@@ -0,0 +1,106 @@
+package datawave.query.data.parsers;
+
+import org.apache.accumulo.core.data.ByteSequence;
+import org.apache.accumulo.core.data.Key;
+
+/**
+ * A reusable, dedicated key parser for Field Index keys
+ * <p>
+ * A Field Index key's components are
+ * <ul>
+ * <li>row - the shard</li>
+ * <li>column family - "fi\0FIELD_NAME"</li>
+ * <li>column qualifier - "value\0datatype\0uid"</li>
+ * </ul>
+ */
+public class FieldIndexKey {
```

Comment:
I'd rather see schematic classes like this somewhere common to ingest and query, but we can always move it later.

Reply:
Sounds like a plan